index.js

```diff
@@ -272,8 +272,8 @@ class Ky {
 		this.request = new globals.Request(this._input, this._options);
 
 		if (this._options.searchParams) {
-			const url = new URL(this.request.url);
```

Comment:
Sorry I'm late to the party. Does this change anything? `this.request.url` is an absolute URL anyway... I'm just asking because I don't like RegExps.

Reply:
I don't like regex either. `this.request.url` can be a relative URL in some environments, e.g. Node and Deno.

Reply:
Then why not this?

```js
const requestUrl = this.request.url;
const searchIndex = requestUrl.indexOf('?');
const urlWithoutParameters = searchIndex === -1 ? requestUrl : requestUrl.slice(0, searchIndex);
const url = `${urlWithoutParameters}${searchParams.toString()}`;
```

Reply:
I believe your code would remove the hash fragment from a URL, unlike the `new URL()` code we had before.

Granted, the hash fragment shouldn't be sent to the server anyway, so that shouldn't really matter. But I figure we should be spec compliant and unobtrusive by only modifying the search params and nothing more. It's possible that a custom `fetch` function might care about the hash fragment, for example.

Reply:
Hmm... Fair point. We could also `indexOf('#')` but that would add more complexity. The regex solution is much clearer.

Reply:
Yep, my thoughts exactly. I did add tests that assert we don't modify the hash fragment. So if someone comes up with a clean non-regex solution, go ahead and give it a shot. It'll be checked thoroughly by the tests.